src/features/list-view/ListView.js

```diff
@@ -38,12 +49,17 @@ class ListView extends React.PureComponent<Props> {
             );
         }
         if (columnIndex === 0) {
+            const { cellData: nameCellData, extension } = getGridCell({ columnIndex, rowIndex });
             return (
                 <div className="list-view-name-cell" key={key} style={style}>
-                    {cellData}
+                    <div className="list-view-icon">
+                        <IconItem className="list-view-icon-item" height={25} iconType={extension} width={25} />
```

Comment:
Let's hoist file-icons up and out and keep ListView dumb i.e. it renders what EUA tells it to render. This also have the benefit of simplifying the new complexity added to your tests.